eq-author/src/App/questionConfirmation/Design/index.js

```diff
@@ -27,14 +29,31 @@ export class UnwrappedQuestionConfirmationRoute extends React.Component {
     onDeleteQuestionConfirmation: PropTypes.func.isRequired,
   };
 
-  handleDeleteClick = () => {
-    this.props.onDeleteQuestionConfirmation(
-      this.props.data.questionConfirmation
+  state = {
+    showDeleteConfirmDialog: false,
+  };
+
+  handleOpenDeleteConfirmDialog = () =>
+    this.setState({ showDeleteConfirmDialog: true });
+
+  handleCloseDeleteConfirmDialog = cb =>
+    this.setState(
+      { showDeleteConfirmDialog: false },
+      isFunction(cb) ? cb : null
     );
+
+  handleDeletePageConfirm = () => {
+    this.handleCloseDeleteConfirmDialog(() => {
```

Comment:
Could we just call setState as the function doesn't seem to do a lot.

Reply:
I can set state in the separate functions. 👍